Package.swift

```diff
@@ -22,4 +24,6 @@ if getenv("SWIFT_SYNTAX_BUILD_SCRIPT") == nil {
   package.products.append(.library(name: "SwiftSyntax", targets: ["SwiftSyntax"]))
 } else {
   package.products.append(.library(name: "SwiftSyntax", type: .dynamic, targets: ["SwiftSyntax"]))
-}
\ No newline at end of file
+}
+
+package.products.append(.library(name: "SwiftSyntaxBuilder", targets: ["SwiftSyntaxBuilder"]))
```

Comment:
Could you put this alone with `SwiftSyntax`? We'd like it to be a dynamic library when building from build script and a static library otherwise.

Reply:
Should we have this functionality as part of the `SwiftSyntax` library instead of a separate `SwiftSyntaxBuilder` library? It will simplify the clients since they won't have to import two separate modules.

Reply:
Since Swift doesn't yet permit dead-stripping anything that's declared public in external modules, adding more code to a module forces its binary size increase on everyone, whether the new APIs are used or not.

So in a case like this, where we have something that's adding value as a layer on top of the underlying library, but isn't critical functionality in and of itself, I think it makes sense to keep it a separate module that folks opt in to.

Reply:
To clarify, if a library is statically linked with an executable, won't SwiftPM do dead-strip on it and remove public APIs that are not used in the executable? \cc @aciidb0mb3r

Reply:
Unless the situation has changed since I last looked into it, anything declared as `open` or `public` is lowered by IRGen as `@llvm.used`, preventing it from ever being dead-stripped by the linker ([SR-1021](https://bugs.swift.org/browse/SR-1021)). That makes sense in a dylib/framework situation where you don't know what the eventual consumer is using, but ends up also applying to statically linked executables.

Reply:
Ah interesting, I wasn't aware of that.

Reply:
I was just not sure I should do so; did in bd31ddb7a032868703f47b4500ae613dfd5bae41.

Reply:
Let's leave it as a separate library while it's still in flux, then we can revisit if we want it to become SwiftSyntax API later.